### PR TITLE
feat(bpdm): post endpoint to upload business partner data using csv file

### PR DESCRIPTION
**Pull request by SujitMBRDI**
## Description

This pull request contains changes for new POST api endpoint on BPDM Gate service, which will accept the CSV file as input for upsetting business partner input data. This endpoint enable way for user to create/update business partner using file.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by SujitMBRDI** [comment]
@nicoprow,
In general, I tried covering almost all necessary changes. 
Just except,  REST API conformity with permissions about new resource and will also adjust default realm files for both helm chart and test.

**Comment by SujitMBRDI** [comment]
> @nicoprow, In general, I tried covering almost all necessary changes. Just except, REST API conformity with permissions about new resource and will also adjust default realm files for both helm chart and test.

@nicoprow, I have adapted changes this changes. Kindly review the changes made.

**Review by nicoprow — CHANGES_REQUESTED** [review]
Since we add a new feature here we should update the POM version to 6.1.0-SNAPSHOT. Likewise, this version upgrade should be written in the Charts as well as described in a comment below

**Comment by SujitMBRDI** [comment]
> Since we add a new feature here we should update the POM version to 6.1.0-SNAPSHOT. Likewise, this version upgrade should be written in the Charts as well as described in a comment below

I rebased current branch with main, so now just updated chart version to 5.0.3-SNAPSHOT as you already created SNAPSHOT versions.

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/model/PartnerUploadFileRow.kt [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdm.gate.model
+
+import com.opencsv.bean.CsvBindByName
+import jakarta.validation.constraints.NotEmpty
+
+data class PartnerUploadFileRow(
+
+    @CsvBindByName(column = "externalId")
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/model/PartnerUploadFileRow.kt]: Done

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/util/PartnerFileUtil.kt [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                nameParts = listOfNotNull(row.nameParts1?.takeIf { it.isNotEmpty() }, row.nameParts2?.takeIf { it.isNotEmpty() }, row.nameParts3?.takeIf { it.isNotEmpty() }, row.nameParts4?.takeIf { it.isNotEmpty() }),
+                identifiers = listOf(
+                    BusinessPartnerIdentifierDto(
+                        type = row.identifiersType ?.takeIf { it.isNotEmpty() },
+                        value = row.identifiersValue ?.takeIf { it.isNotEmpty() },
+                        issuingBody = row.identifiersIssuingBody ?.takeIf { it.isNotEmpty() }
+                    )
+                ),
+                states = listOf(BusinessPartnerStateDto(
+                    validFrom = row.statesValidFrom?.let { LocalDateTime.parse(it, formatter) },
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/util/PartnerFileUtil.kt]: Adapted with new function

File: bpdm-gate/src/test/kotlin/org/eclipse/tractusx/bpdm/gate/controller/PartnerUploadControllerIT.kt [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import java.nio.file.Files
+import java.nio.file.Paths
+
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@ActiveProfiles("test-no-auth")
+@ContextConfiguration(initializers = [PostgreSQLContextInitializer::class])
+class PartnerUploadControllerIT @Autowired constructor(
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply bpdm-gate/src/test/kotlin/org/eclipse/tractusx/bpdm/gate/controller/PartnerUploadControllerIT.kt:57]: Created new test which covers check for the expectation on created business partner input data.

File: charts/bpdm/Chart.yaml [hunk 4]
```diff
@@ -22,7 +22,7 @@ apiVersion: v2
 name: bpdm
 type: application
 description: A Helm chart for Kubernetes that deploys the BPDM applications
-version: 5.0.1
+version: 5.0.2
 appVersion: "6.0.1"
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply charts/bpdm/Chart.yaml]: Done

File: bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/model/response/PartnerUploadErrorResponse.kt [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import java.time.Instant
+
+@Schema(description = "Error response for invalid partner upload")
+class PartnerUploadErrorResponse(
+    @Schema(description = "Timestamp of the error occurrence")
+    val timestamp: Instant,
+    @Schema(description = "HTTP status of the error response")
+    val status: HttpStatus,
+    @Schema(description = "List of error messages")
+    val error: List<String>,
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/model/response/PartnerUploadErrorResponse.kt:33]: As mentioned on sig release issue [700](https://github.com/eclipse-tractusx/sig-release/issues/700).
More information may come from Juila so, for now i haven't refactored this much.But yes in future we can refactor this. Keeping this for future scope, hope you agree with this.

File: bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/GatePartnerUploadApi.kt [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        value = [
+            ApiResponse(responseCode = "200", description = "Business partners were successfully updated or created"),
+            ApiResponse(responseCode = "400", description = "On malformed Business partner upload request",
+                content = [Content(
+                    mediaType = "application/json",
+                    schema = Schema(implementation = PartnerUploadErrorResponse::class)
+                )]),
+        ]
+    )
+    @PostMapping("/input/partner-upload-process/upload-partner-csv", consumes = ["multipart/form-data"])
```
[the remark this answers is not in the document]
Comment (SujitMBRDI) [reply bpdm-gate-api/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/api/GatePartnerUploadApi.kt]: Done

**Comment by nicoprow** [comment]
> > Since we add a new feature here we should update the POM version to 6.1.0-SNAPSHOT. Likewise, this version upgrade should be written in the Charts as well as described in a comment below
> 
> I rebased current branch with main, so now just updated chart version to 5.0.3-SNAPSHOT as you already created SNAPSHOT versions.

I the current app version is 6.0.1-SNAPSHOT. Since this pull request adds a new feature to the app, we need to increase the SNAPSHOT version to 6.1.0-SNAPSHOT. Likewise, we need to increase the chart's SNAPSHOT version to 5.1.0-SNAPSHOT as well as update its appVersion to the 6.1.0-SNAPSHOT.

**Comment by SujitMBRDI** [comment]
> > > Since we add a new feature here we should update the POM version to 6.1.0-SNAPSHOT. Likewise, this version upgrade should be written in the Charts as well as described in a comment below
> > 
> > 
> > I rebased current branch with main, so now just updated chart version to 5.0.3-SNAPSHOT as you already created SNAPSHOT versions.
> 
> @SujitMBRDI I the current app version is 6.0.1-SNAPSHOT. Since this pull request adds a new feature to the app, we need to increase the SNAPSHOT version to 6.1.0-SNAPSHOT. Likewise, we need to increase the chart's SNAPSHOT version to 5.1.0-SNAPSHOT as well as update its appVersion to the 6.1.0-SNAPSHOT.

@nicoprow, Done. I have adapted changes.

**Review by nicoprow — APPROVED** [review]
Looks good to me

**Pull request merged**